src/Adapter/Product/CommandHandler/UpdateProductBasicInformationHandler.php

```diff
@@ -110,7 +110,7 @@ private function fillUpdatableProperties(
             // Go through all the product languages and make sure name is filled for each of them
             $productLanguages = array_keys($product->name);
             foreach ($productLanguages as $languageId) {
-                if (empty($product->name[$languageId]) && empty($localizedNames[$languageId])) {
+                if (empty($localizedNames[$languageId])) {
```

Comment:
now we always just check if its trying to empty the name (if yes, we use default, no matter if product had the name before or not)